test/test-ParametersParser.cc

```diff
@@ -84,7 +84,7 @@ TEST_CASE("Parse illegal parameter among legal ones", "[parameters]")
     "+ Parameters\n"\
     "    + OK_1\n"\
     "    + i;legal\n"\
-    "    + OK-2\n";
+    "    + OK_2\n";
```

Comment:
With this PR, whenever we find ambiguous parameters, we will be defaulting to new syntax assumption.

In the new MSON aligned syntax `OK-2` is `OK` parameter name and `2` description. So, I edited it to `OK_2` so that `OK_2` is parameter name.